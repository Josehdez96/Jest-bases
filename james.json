{
  "id": 1,
  "name": "James Rodriguez",
  "status": "Active",
  "species": "Human",
  "gender": "Male"
}
