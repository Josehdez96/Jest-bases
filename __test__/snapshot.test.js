import { getCharacter } from '../snapshot';
import james from '../james.json';

describe('Probando los snapshots', () => {
  test('Snapshot de James', () => {
    expect(getCharacter(james)).toMatchSnapshot();
  });

  test('Siempre fallará el snapshot', () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20),
    };
    expect(user).toMatchSnapshot();
  });

  test('Tenemos una excepcion', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: 'James Juan',
    };
    expect(user).toMatchSnapshot({
      //Aquí va la excepcion
      id: expect.any(Number),
    });
  });
});
