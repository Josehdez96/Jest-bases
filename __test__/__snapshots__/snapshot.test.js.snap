// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando los snapshots Siempre fallar√° el snapshot 1`] = `
Object {
  "createAt": 2020-09-09T17:19:40.882Z,
  "id": 4,
}
`;

exports[`Probando los snapshots Snapshot de James 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "James Rodriguez",
  "species": "Human",
  "status": "Active",
}
`;

exports[`Probando los snapshots Tenemos una excepcion 1`] = `
Object {
  "id": Any<Number>,
  "name": "James Juan",
}
`;
